plugins/ntuplizer.cc

```diff
@@ -431,6 +505,105 @@ void ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
     // lowPtElecs.avgClusterZ.push_back(tcz/nbsc_seed);
    }
 
+   for (int i=0; i<nLowPtElectron; i++){
+    //Gets first electron
+    std::cout << "Electron" << std::endl;
+    std::cout << i << std::endl;
+    Lxy_SV = -99;
+    normalizedChi2 = 0;
+    //hasValidVertex = false;
+    for (int j=i+1; j<nLowPtElectron; j++){
+      std::cout << "Paied with Electron" << std::endl;
+      std::cout << j << std::endl;
+      //isEE = false;
+      double charge1 = lowPtElectrons->at(i).charge();
+      double charge2 = lowPtElectrons->at(j).charge();
+      if (charge1*charge2>0){
+        std::cout << "Invalid Pair" << std::endl;
+        continue;
+      };
+        
+    
+      // Get tracks:
+
+      const reco::Track &tr_A = *(lowPtElectrons->at(i).gsfTrack());
+      const reco::Track &tr_B = *(lowPtElectrons->at(j).gsfTrack());
+      std::vector<reco::TransientTrack> vec_refitTracks;
+      reco::TransientTrack isotransienttrackA = theB->build(tr_A);
+      reco::TransientTrack isotransienttrackB = theB->build(tr_B);
+      vec_refitTracks.push_back(isotransienttrackA); vec_refitTracks.push_back(isotransienttrackB);
+
+      // Fit tracks:
+      KalmanVertexFitter thefitterll;
+      TransientVertex myVertex = thefitterll.vertex(vec_refitTracks);
+      const reco::Vertex secV = myVertex;
+      if (secV.isValid()) {
+      
+        //hasValidVertex = true;
+
+        // Define the axis along the direction of the distance is defined:
+        GlobalVector axis(0,0,0);
+        axis = GlobalVector(secV.x(),secV.y(),secV.z());
+        /*
+        // Define the errors and points of the CMS centre point and the beam spot:
+        math::Error<3>::type e0; // dummy
+        math::Error<3>::type cov = bs.covariance3D();
+        math::XYZPoint pbs(bs.x0(), bs.y0(), bs.z0());
+        math::XYZPoint p0(0.0, 0.0, 0.0);
+
+        // Fake vertices for the CMS centre and beam spot:
+        const reco::Vertex v0(p0, e0);
+        const reco::Vertex vbs(pbs, cov);
+
+        // Measurements:
+        Measurement1D vMeas_PV = reco::SecondaryVertex::computeDist2d(pv,secV,axis,true);
+        Measurement1D vMeas_0 = reco::SecondaryVertex::computeDist2d(v0,secV,axis,false);
+        Measurement1D vMeas_BS = reco::SecondaryVertex::computeDist2d(vbs,secV,axis,true);
+        
+
+        // Distance values:
+        Lxy_0 = vMeas_0.value();
+        Ixy_0 = vMeas_0.significance();
+        Lxy_PV = vMeas_PV.value();
+        Ixy_PV = vMeas_PV.significance();
+        Lxy_BS = vMeas_BS.value();
+        Ixy_BS = vMeas_BS.significance();
+
+        */// Vertex position and fit details:
+        temp_normalizedChi2 = myVertex.normalisedChiSquared();         
+        vx = secV.x();
+        vy = secV.y();
+        temp_Lxy_SV = std::sqrt(std::pow(vx-PVs.x, 2) + std::pow(vy-PVs.y, 2));
+        std::cout << "temp_Lxy_SV" << std::endl;
+        std::cout << temp_Lxy_SV << std::endl;
+        if (temp_normalizedChi2 > normalizedChi2){
```

Comment:
We should keep the ones minimizing the Chi2/ndof, not maximizing, so > should be <